bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampHandler.java

```diff
@@ -0,0 +1,247 @@
+
+    long refreshInterval = 30;
+    static boolean isEsoAvailable = false;
+    static boolean isEsmAvailable = false;
+
+    public FerroampHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String transId = UUID.randomUUID().toString();
+        String valueConfiguration = command.toString();
+        if (FerroampBindingConstants.CHANNEL_REQUESTCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"charge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonCharge, ferroampConfig);
+        }
+        if (FerroampBindingConstants.CHANNEL_REQUESTDISCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonDisCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"discharge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonDisCharge, ferroampConfig);
+        }
+        if (FerroampBindingConstants.CHANNEL_AUTO.equals(channelUID.getId())) {
+            String requestCmdJsonAuto = "{\"" + "transId" + "\":\"" + transId + "\",\"cmd\":{\"name\":\"auto\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonAuto, ferroampConfig);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Set channel configuration parameters
+        channelConfigEhub = FerroampChannelConfiguration.getChannelConfigurationEhub();
+        channelConfigSsoS1 = FerroampChannelConfiguration.getChannelConfigurationSsoS1();
+        channelConfigSsoS2 = FerroampChannelConfiguration.getChannelConfigurationSsoS2();
+        channelConfigSsoS3 = FerroampChannelConfiguration.getChannelConfigurationSsoS3();
+        channelConfigSsoS4 = FerroampChannelConfiguration.getChannelConfigurationSsoS4();
+        channelConfigEso = FerroampChannelConfiguration.getChannelConfigurationEso();
+        channelConfigEsm = FerroampChannelConfiguration.getChannelConfigurationEsm();
+
+        final MqttBrokerConnection ferroampConnection = new MqttBrokerConnection(ferroampConfig.hostName,
+                FerroampBindingConstants.BROKER_PORT, false, false, ferroampConfig.userName);
+
+        scheduler.scheduleWithFixedDelay(this::pollTask, 60, refreshInterval, TimeUnit.SECONDS);
+        this.setFerroampConnection(ferroampConnection);
+    }
+
+    private void pollTask() {
+        try {
+            startMqttConnection();
+        } catch (InterruptedException e) {
+            logger.debug("Problems with startMqttConnection()");
+        }
+
+        Objects.requireNonNull(ferroampConnection, "MqttBrokerConnection ferroampConnection cannot be null");
+        if (ferroampConnection.connectionState().toString().equals("DISCONNECTED")) {
```

Comment:
The requireNonNull can be used to 'convert' a nullable variable to a non-null.
In this case you can use a local variable. Using a local variable fixes this issue as a local variable cannot be cahnged between the check and the usages. A variable at a wider scope (class level) could be changed by a different thread in between the if check and the usage. I know this is a race condition that actually never happens, but it _can_ happen so the compiler complains about it and that is why we need to fix it.
```suggestion
        FerroampConnection ferroampConnection = this.ferroampConnection;
        if (ferroampConnection == null || ferroampConnection.connectionState().toString().equals("DISCONNECTED")) {
```

Reply:
20241125
Still struggling with this. I think I understand what you describe but I have still not figure out how to solve it.
And I don't  understand about line 110, FerroampConnection ferroampConnection = this.ferroampConnection; should I make a new Class FerroampConnection?
I have so far used Class MqttBrokerConnection for ferroampConnection

Reply:
20241126
Please see my latest changes in FerroampHandler.java
Changes done in:
method initialize, line 97
method pollTask, line 113
method startMqttConnection, line 136

Reply:
20241128
Please see my latest changes in FerroampHandler.java and FerroampMqttCommunication.java
There are at least now no warnings left ;-)